### PR TITLE
Enable sync all and add selection type tracking

### PR DESCRIPTION
**Pull request by artemruts**
### Description

This PR adds the following functionality:
1. enables the `sync all` UI for the "manage repositories" and post-sign-up flow pages
2. repositories now have `beta` badges instead of `new`
3. adds tracking to learn if the user is using the `sync all` feature or selects individual (own) repositories
4. adds tracking to learn if the user manually added public repositories by URL

#### "manage repositories"  page

**event name**: `UserManageRepositoriesSave`
**payload**:

```ts
 {
     "userReposSelection": "all" | "specific" | null,
     "didAddReposByURL": boolean
 }
```
####  post-sign-up flow

**event name**: `PostSignUp_Repos_Saved`
**payload**:

```ts
 {
     "userReposSelection": "all" | "specific" | null,
 }
```

### UI changes

**Beta badges**
beta badges

**Link to `sync all` docs**
sync all docs

### CODE REVIEW

**Review by quinnkeast — CHANGES_REQUESTED** [review]
We should avoid sprinkling links to docs at too granular a level, as then we're using the docs as a crutch instead of making the UI itself self-explanatory and intuitive.

With that in mind, let's remove the link from the helper text in the "Sync all" option, and update the top-level messaging to include a learn more link, which is a pattern we use in other places in Sourcegraph, and point at the [general docs page](https://docs.sourcegraph.com/code_search/how-to/adding_repositories_to_cloud rather than specifically the "Sync all" section.

Here's a screenshot of those changes (slight wording adjustment):

image

>Choose which repositories to sync with Sourcegraph to search all your code in one place. <a>Learn more</a>

Additionally, I'm not sure if this is a consequence of this PR or something else, but the "other repositories by URL" option is missing:

image

**Comment by artemruts** [comment]
> We should avoid sprinkling links to docs at too granular a level, as then we're using the docs as a crutch instead of making the UI itself self-explanatory and intuitive.
> 
> With that in mind, let's remove the link from the helper text in the "Sync all" option, and update the top-level messaging to include a learn more link, which is a pattern we use in other places in Sourcegraph, and point at the [general docs page](https://docs.sourcegraph.com/code_search/how-to/adding_repositories_to_cloud) rather than specifically the "Sync all" section.
> 
> Here's a screenshot of those changes (slight wording adjustment):
> 
> image
> 
> > Choose which repositories to sync with Sourcegraph to search all your code in one place. Learn more
> 
> Additionally, I'm not sure if this is a consequence of this PR or something else, but the "other repositories by URL" option is missing:
> 
> image

Thank you, I just didn't see any notes on where the link should go and to what section should it link. That's why I went with ☝️

**Comment by artemruts** [comment]
> Additionally, I'm not sure if this is a consequence of this PR or something else, but the "other repositories by URL" option is missing:
> 
> image

I just checked locally, and the section is there. Maybe it has to do with how you're running the local instance if you use `sg.

File: client/web/src/auth/PostSignUpPage.tsx [hunk 1]
```diff
@@ -168,9 +168,17 @@ export const PostSignUpPage: FunctionComponent<PostSignUpPage> = ({
                                             <div className="mt-5">
                                                 <h3>Add repositories</h3>
                                                 <p className="text-muted mb-4">
-                                                    Choose repositories you own or collaborate on from your code hosts
-                                                    to search with Sourcegraph. We’ll sync and index these repositories
-                                                    so you can search your code all in one place.
+                                                    Choose repositories you own or collaborate on from your code hosts.
+                                                    We’ll sync and index these repositories so you can search your code
```
Comment (artemruts) [inline client/web/src/auth/PostSignUpPage.tsx:172]: @quinnkeast I removed `to search with Sourcegraph`, I think it's obvious and will make lines break better.

File: client/web/src/auth/PostSignUpPage.tsx [hunk 2]
```diff
@@ -168,9 +168,17 @@ export const PostSignUpPage: FunctionComponent<PostSignUpPage> = ({
...
+                                                    all in one place.
+                                                    <Link
+                                                        to="https://docs.sourcegraph.com/code_search/how-to/adding_repositories_to_cloud"
+                                                        target="_blank"
+                                                        rel="noopener noreferrer"
+                                                    >
+                                                        {' '}
```
Comment (5h1rU) [inline client/web/src/auth/PostSignUpPage.tsx:179]: Do you need this empty space?

**Review by 5h1rU — APPROVED** [review]
Excellent description. 🔥

File: client/web/src/auth/PostSignUpPage.tsx (hunk 2 above)
Comment (artemruts) [reply client/web/src/auth/PostSignUpPage.tsx:179]: I do 😄

**Review by quinnkeast — APPROVED** [review]
Looks great! Thanks!

File: client/web/src/user/settings/repositories/UserSettingsManageRepositoriesPage.tsx [hunk 3]
```diff
@@ -455,12 +452,20 @@ export const UserSettingsManageRepositoriesPage: React.FunctionComponent<Props>
...
+
+            const loggerPayload = {
+                userReposSelection: selectionState.radio
+                    ? selectionState.radio === 'selected'
+                        ? 'specific'
+                        : 'all'
+                    : null,
+                didAddReposByURL: !!publicRepos.length,
             }
+            eventLogger.log('UserManageRepositoriesSave', loggerPayload, loggerPayload)
```
Comment (attfarhan) [inline client/web/src/user/settings/repositories/UserSettingsManageRepositoriesPage.tsx:468]:
```suggestion
            eventLogger.log('UserSettingsManageRepositoriesSaved', loggerPayload, loggerPayload)
```

Comment (attfarhan) [reply client/web/src/user/settings/repositories/UserSettingsManageRepositoriesPage.tsx:468]: To be consistent with other user settings page events.

**Review by attfarhan — APPROVED** [review]

**Review by ryphil — APPROVED** [review]
Metrics look good. Would love to track if they hit the Skip for Now button if possible but approving to unblock.

File: client/web/src/user/settings/repositories/UserSettingsManageRepositoriesPage.tsx (hunk 3 above)
Comment (artemruts) [reply client/web/src/user/settings/repositories/UserSettingsManageRepositoriesPage.tsx]: Thank you!

**Comment by artemruts** [comment]
> Metrics look good. Would love to track if they hit the Skip for Now button if possible but approving to unblock.

Synced with Ryan offline. "Skip" metrics already went out with https://github.com/sourcegraph/sourcegraph/pull/24022

**Pull request merged**